### DISCUSSION TITLE
Kernel Compile problems

### DISCUSSION THREAD

**Discussion by Wolke926** [idaholab/moose#25066] [Q&A Getting Started] [answered]
I am studying the "Babbler" example and trying to add some new kernels in this application, I followed the steps here:
https://mooseframework.inl.gov/getting_started/examples_and_tutorials/tutorial01_app_development/step05_kernel_object.html

I put the .c and .h files in the src and include folder respectively, then run make -j 4. The .C and .h files I want to add are just copied from modules files. I also changed the App name for .C file.
<img width="414" alt="Screenshot 2023-07-31 at 2 29 37 PM" src="https://github.com/idaholab/moose/assets/120539827/46070773-8003-4e23-a331-20121f0348a5">

But when I want to run " cd ~/projects/babbler/problems
../babbler-opt -i my_own-project.i  " the kernel I want to add are still not been registered.
![Screenshot 2023-07-31 at 2 32 56 PM](https://github.com/idaholab/moose/assets/120539827/6dbe8d08-fad8-4a78-b487-56763c4e6e36)
Are there any steps missing?

**Comment by lindsayad** [comment, marked as answer]
If you want to use classes from a module, you should modify the Makefile in your application, toggling `PHASE_FIELD := yes`

But with respect to your error, you showed that you registered ADSplitCHWRes, but the error is about the non-AD object SplitCHWRes. Did you register the non-AD version to your app?

**Comment by Wolke926** [comment]
I changed the makefile then compile again, but I got this:
<img width="726" alt="Screenshot 2023-08-01 at 9 22 47 AM" src="https://github.com/idaholab/moose/assets/120539827/2edce455-73cf-46e8-a913-7dff0dfaa2ba">

**Reply by lindsayad** [reply]
Do you still have the copies in your application? I would make sure you've remove the copies and you're only using the phase field module version of that kernel

**Reply by Wolke926** [reply]
I remove all the moose files and download them again, I guess there are some problems in the source file. Now It works.
Thank you.
